Makefile

```diff
@@ -335,11 +335,17 @@ mockgen:
 
 ### Docker images ###
 
+# This target is for development only. It assumes that "make bin" has been run previously and will
+# copy the local binaries to the Docker image, instead of building the binaries inside the image as
+# part of the Docker build.
+
 .PHONY: ubuntu
 ubuntu:
```

Comment:
A nit, Can we remove ubuntu, and we can modify the CONTRIBUTING.md file to make use of `make build-agent-ubuntu` and `make build-controller-ubuntu` instead of `make ubuntu`.

Reply:
These targets do different things. I know that some folks prefer to build the binaries externally and copy them inside the Docker container (building this way is less portable but may be faster).

Reply:
okay got it.